custom/enikshay/private_sector_datamigration/management/commands/update_user_contact_phone_number.py

```diff
@@ -0,0 +1,45 @@
+from django.core.management import BaseCommand
+
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.users.models import CommCareUser
+from corehq.util.log import with_progress_bar
+from custom.enikshay.private_sector_datamigration.models import UserDetail
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('location_ids', nargs='*')
+
+    def handle(self, domain, location_ids, **options):
+        locs_matching_type = SQLLocation.active_objects.filter(
+            domain=domain,
+            location_type__code__in=[
+                'pac',
+                'pcp',
+                'plc',
+                'fo',
+                'pcc',
+            ],
+        )
+        if location_ids:
+            locs_matching_type = locs_matching_type.filter(location_id__in=location_ids)
+
+        for _, loc in enumerate(with_progress_bar(locs_matching_type)):
+            private_sector_agency_id = loc.metadata.get('private_sector_agency_id', '')
+            if not private_sector_agency_id.isdigit():
+                continue
+            agency_id = int(private_sector_agency_id)
+
+            try:
+                user_detail = UserDetail.objects.get(agencyId=agency_id, isPrimary=True)
+            except UserDetail.DoesNotExist:
+                continue
+
+            location_user = CommCareUser.get_by_user_id(loc.user_id)
+
+            contact_phone_number = location_user.user_data.get('contact_phone_number')
+            if not contact_phone_number and len(user_detail.mobileNumber) == 10:
```

Comment:
are the phone numbers on the user detail already cleaned? otherwise we should probably clean any non digit characters before checking for length.

also we can handle when the source number is 11 digits long and it starts with a leading zero by just removing the leading zero.

and do we know that these are always local numbers, or could some of them already be 12 digits with 91 in front?

Reply:
> are the phone numbers on the user detail already cleaned?

yes

> also we can handle when the source number is 11 digits long and it starts with a leading zero by just removing the leading zero.

All phone numbers here have either 10 or 6 digits.  I'm skipping the ones with 6 digits.